### PR TITLE
Add function on vehicle to create a command URL

### LINKED ISSUE #35: Construct URLs with functions
i.e. `path.Join(u.Path, "vehicles", strconv.FormatUint(id, 10), "vehicle_data")`

### PR DESCRIPTION
**Pull request by bogosj**
This cleanup resolves #35.

### CODE REVIEW

**Comment by bogosj** [comment]
Hold off on reviewing, I'm going to re-do the tests with Convey.

File: vehicles.go [hunk 1]
```diff
@@ -91,3 +92,14 @@ func (c *Client) Vehicle(vehicleID int64) (*Vehicle, error) {
 	resp.Response.c = c
 	return resp.Response, nil
 }
+
+// CommandPath returns the URL for a provided command.
+func (v *Vehicle) CommandPath(command string) string {
```
Comment (michaelharo) [inline vehicles.go:97]: Does `CommandPath` need to be exported?

File: states.go [hunk 2]
```diff
@@ -348,8 +349,8 @@ func stateError(sr *StateRequest) error {
 // A utility function to fetch the appropriate state of the vehicle
 func (c *Client) fetchState(resource string, id int64) (*StateRequest, error) {
 	stateRequest := &StateRequest{}
-	if err := c.getJSON(
-		c.baseURL+"/vehicles/"+strconv.FormatInt(id, 10)+"/data_request"+resource, stateRequest); err != nil {
+	path := strings.Join([]string{c.baseURL, "vehicles", strconv.FormatInt(id, 10), "data_request", resource}, "/")
```
Comment (michaelharo) [inline states.go:352]: `path.Join` ?

File: vehicles.go [hunk 3]
```diff
@@ -91,3 +92,14 @@ func (c *Client) Vehicle(vehicleID int64) (*Vehicle, error) {
...
+	// The wake command is not rooted at .../command.
+	parts := []string{v.c.baseURL, "vehicles", strconv.FormatInt(v.ID, 10)}
+	if command != "wake_up" {
```
Comment (michaelharo) [inline vehicles.go:100]: This seems a bit unexpected here.  Perhaps there should be a couple different functions

1. Something that provides the base path (for use by the next 2 and also wake_up?)
2. Something that works for commands
3. Something that works for fetching state (like already exists)

**Review by michaelharo — CHANGES_REQUESTED** [review]

File: vehicles.go (hunk 3 above)
Comment (andig) [reply vehicles.go:100]: I‘d probably just not use commandPath for wakeup then and add a comment to that effect

File: states.go (hunk 2 above)
Comment (bogosj) [reply states.go:352]: I was using path.Join, but it was doing something weird in tests. It seems to strip the double / from the protocol? Didn't seem "right" so I used strings.Join.

https://play.golang.org/p/EEF_iUtUIGK

**Comment by bogosj** [comment]
PTAL

**Comment by andig** [comment]
Cumbersome: https://stackoverflow.com/questions/34668012/combine-url-paths-with-path-join

**Comment by bogosj** [comment]
> Cumbersome: https://stackoverflow.com/questions/34668012/combine-url-paths-with-path-join

Indeed that method seems unnecessary, I'm fine leaving this with strings.Join

**Comment by bogosj** [comment]
Ping @michaelharo, this look good for now?

**Comment by michaelharo** [comment]
> Ping @michaelharo, this look good for now?

I'm fine with this for now.

**Review by michaelharo — APPROVED** [review]

**Pull request merged**